code/game/turfs/simulated/floor_types.dm

```diff
@@ -140,25 +141,68 @@
 			secured = !secured
 			C.playtoolsound(src, 80)
 			update_icon()
+			
+/turf/simulated/floor/engine/proc/explode_layers(var/layers = 1)
+	if(secured)		//plasteel tile, screwed in
+		secured = FALSE
+		update_icon()
+		layers -= 1
+		if(!layers)
+			return
+	if(floor_tile)	//plasteel tile, unscrewed
+		qdel(floor_tile)
+		floor_tile = null
+		icon_state = "engine"
+		update_icon()
+		layers -= 1
+		if(!layers)
+			return
+	//reinforced floor
+	new /obj/item/stack/rods(src, 1)
+	ChangeTurf(/turf/simulated/floor)
+	var/turf/simulated/floor/F = src
+	F.make_plating()
+	layers -= 1
+	
+	//normal plating
+	if(!layers)
+		return
+	var/obj/structure/cable/C = locate(/obj/structure/cable/) in loc
+	var/obj/machinery/atmospherics/pipe/P = locate(/obj/machinery/atmospherics/pipe) in loc
+	var/obj/structure/disposalpipe/D = locate(/obj/structure/disposalpipe) in loc
```

Comment:
I don't think `in loc` is what you meant to put here, since the turf is `src`, not `loc`

Reply:
doesn't loc refer to src?
i just copypasted the `locate() in loc `from somewhere and it seems to work
if it works, don't fix it, unless you really want me to

Reply:
A turf's `loc` is the area it is a part of. Only the turfs are included in the area's `contents`, so these `locate()`s will never find anything
It's also worth noting that even if you were locating in `src`, it would only ever damage one of each of these objects per turf. Should probably loop through, instead